pennylane/ops/op_math/prod.py

```diff
@@ -315,14 +317,21 @@ def eigvals(self):
     def matrix(self, wire_order=None):
```

Comment:
@Jaybsoni out of curiosity, since wire order is a argument to this function, could you just use `@functools.lrucache` rather than doing it manually?

(or alternatively, `cachetools` which provides more configuration)

Reply:
I remember looking into using `lrucache` in the past, but there were some issue with using it. I can look into it again.